components/x-teaser-timeline/src/TeaserTimeline.jsx

```diff
@@ -6,6 +6,21 @@ import { getDateOnly } from './lib/date'
 import styles from './TeaserTimeline.scss'
 import classNames from 'classnames'
 
+/**
+ * @typedef {Object} Props
+ * @property {string} csrfToken
+ * @property {boolean} showSaveButtons
+ * @property {Array<Object> | Array<string> | Object | string} customSlotContent
+ * @property {Array<number> | number} customSlotPosition
+ * @property {Array<Item>} items
+ * @property {number} timezoneOffset
+ * @property {string} localTodayDate
+ * @property {string} latestItemsTime,
+ * @property {number} latestItemsAgeHours
+ *
+ * @param {Props} props
+ * @returns
```

Comment:
```suggestion
 * @returns {(props: Props) => JSX.Element}
```

Suggestion [non-blocking]: I'd always recommend explicitly typing the return signature: it guards against refactors inadvertently changing the behaviour of the method

(a classic one is when an arrow function is refactored into a plain function declaration and the developer forgets the `return` keyword)